src/farcasterd/runtime.rs

```diff
@@ -311,6 +317,26 @@ impl esb::Handler<ServiceBus> for Runtime {
 }
 
 impl Runtime {
+    fn services_ready(&self) -> Result<(), Error> {
+        if !self.walletd_ready {
+            Err(Error::Farcaster(
+                "Farcaster not ready yet, walletd still starting".to_string(),
+            ))
+        } else if !self.databased_ready {
+            Err(Error::Farcaster(
+                "Farcaster not ready yet, databased still starting".to_string(),
+            ))
+        } else {
+            Ok(())
+        }
+    }
+    fn peer_keys_ready(&self) -> Result<(SecretKey, PublicKey), Error> {
+        if self.node_secret_key.is_some() && self.node_public_key.is_some() {
```

Comment:
I think `if let` with a tuple can work here to avoid `.is_some()` and `.unwrap()`... maybe

Reply:
Yes, done in 0a4bdf167056f416833fdc5dc75b107323cd8260